doc/examples/tutorial_ibl_choice_decoding.md

```diff
@@ -0,0 +1,180 @@
+
+For a more detailed tutorial on data loading with DANDI, see the "Streaming data from DANDI" example!
+
+**Caveats**: This example is meant to provide a simple starting point for working with trial-aligned data and data from the IBL, and so it does not faithfully replicate the IBL's quality control and filtering criteria; the decoding here is also simpler than the analyses carried out in those works.
+
+```{code-cell} ipython3
+from pathlib import Path
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import pynapple as nap
+import scipy.stats
+
+from sklearn.linear_model import LogisticRegression
+from sklearn.model_selection import RandomizedSearchCV
+```
+
+## Loading data
+
+See also the "Streaming data from DANDI", which includes more detail.
+
+```{code-cell} ipython3
+# These modules are used for data loading
+from dandi.dandiapi import DandiAPIClient
+import fsspec
+from fsspec.implementations.cached import CachingFileSystem
+import h5py
+from pynwb import NWBHDF5IO
+```
+
+```{code-cell} ipython3
+# The BWM dandiset:
+dandiset_id = "000409"
+# This is a randomly chosen recording session.
+asset_path = "sub-CSH-ZAD-026/sub-CSH-ZAD-026_ses-15763234-d21e-491f-a01b-1238eb96d389_behavior+ecephys+image.nwb"
+with DandiAPIClient() as client:
+    asset = client.get_dandiset(dandiset_id, "draft").get_asset_by_path(asset_path)
+    s3_url = asset.get_content_url(follow_redirects=1, strip_query=True)
+fs = CachingFileSystem(
+    fs=fsspec.filesystem("http"),
+    cache_storage=str(Path("~/.caches/nwb-cache").expanduser()),
+)
+io = NWBHDF5IO(file=h5py.File(fs.open(s3_url, "rb")), load_namespaces=True)
+nwb = nap.NWBFile(io.read())
+```
+
+This NWB file, loaded into a pynapple object, contains neural activity, behavioral data, and raw electrophysiological traces. They're lazily loaded, so only what we use will be downloaded.
+
+```{code-cell} ipython3
+nwb
+```
+
+The only fields we'll use here are the neural activity ('units') and the trial information ('trials').
+
+```{code-cell} ipython3
+spikes = nwb['units']
+trials = nwb['trials']
+```
+
+```{code-cell} ipython3
+spikes
+```
+
+```{code-cell} ipython3
+trials
+```
+
+## Trial alignment and binning
+
+Here, we start by discarding some trials where a choice wasn't made (a fuller analysis may include more criteria). Next, we create an `IntervalSet` aligned to the stimulus onset time within each trial, where the window around each stimulus extends back 0.5s and forward 2s.
+
+```{code-cell} ipython3
+valid_choice = trials.choice != 0
+trials = trials[valid_choice]
+stim_on_intervals = nap.IntervalSet(
+    start=trials.stim_on_time - 0.5,
+    end=trials.stim_on_time + 2.0,
+)
+```
+
+Restrict the neural activity to the included trial epochs:
+
+```{code-cell} ipython3
+spikes = spikes.restrict(trials)
```

Comment:
It's not necessary if you do the build_tensor after. The spikes are automatically restricted.